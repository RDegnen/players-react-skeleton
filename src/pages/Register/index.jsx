import React from 'react';
import { FormGroup, Label, Button } from 'reactstrap';
import { withRouter } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';

import constants from '../../constants';
import InputForm from '../../components/InputForm';

const createUser = (data, history) =>
  fetch(`${constants.API_URL}/api/user`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  }).then(res => res.json())
    .then((json) => {
      localStorage.setItem('token', json.token);
      history.push('/roster');
    })
    .catch(err => err);

const formGroups = [
  {
    field: 'first_name',
    type: 'text',
    id: 'firstName',
    label: 'First Name',
  },
  {
    field: 'last_name',
    type: 'text',
    id: 'lastName',
    label: 'Last Name',
  },
  {
    field: 'email',
    type: 'email',
    id: 'email',
    label: 'Email',
  },
  {
    field: 'password',
    type: 'password',
    id: 'password',
    label: 'Password',
  },
  {
    field: 'confirm_password',
    type: 'password',
    id: 'confirmPassword',
    label: 'Confirm Password',
  },
  {
    renderGroup: () => (
      <FormGroup row>
        <Label for="register">Register</Label>
        <Button type="submit" value="Submit" id="register">Submit</Button>
      </FormGroup>
    ),
  },
];

class Register extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleInput = this.handleInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInput({ target }) {
    const { name, value } = target;
    this.setState({ [name]: value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { history } = this.props;
    createUser(this.state, history);
  }
  render() {
    return (
      <InputForm
        formGroups={formGroups}
        handleInput={this.handleInput}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

Register.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};

export default withRouter(Register);
