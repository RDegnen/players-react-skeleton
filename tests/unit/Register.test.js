import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Register from '../../src/pages/Register';

Enzyme.configure({ adapter: new Adapter() });

const getComponent = props => shallow(<Register history={{}} {...props} />);

describe('<Register />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });
});
