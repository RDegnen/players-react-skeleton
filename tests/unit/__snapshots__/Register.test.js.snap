// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
